### ISSUE TITLE
Cannot locate kernel json file when connecting to remote ipython kernel

### ISSUE THREAD

**Issue by mattsqerror** [spyder-ide/spyder#2945]
When attempting to connect to a remote ipython kernel, a dialog box comes up which allows browsing for the applicable `kernel-????.json` file.  After browsing for and selecting the json file (or manually typing in the path and name file), entering the ssh information, and clicking ok, the error "Unable to connect to Ipython `kernel-????`" appears.

After removing the try/except from `_create_client_for_kernel()` function in `ipythonconsole.py`, I discovered that `find_connection_file()` is searching for the json file in the path `/run/user/1000/jupyter`, regardless of the path entered into the dialog box. Commenting the line `cf = osp.basename(cf)`, which gets rid of the path, appears to fix the problem, since the `jupyter_console` function `find_connection_file()` appears to default to looking for the json file in the aforementioned directory. Of course copying the json file to the `/run/user/1000/jupyter` directory also fixes the problem, but that seems unnecessary.

Is this a bug, or am I missing something?

**Comment by ccordoba12** [comment]
Sorry, this was something missing when we started to update Jupyter through the new `qtconsole` package.

It'll be fixed in Spyder 3.0 beta3.

**Comment by Pitje06** [comment]
There's still something wrong with this in version 3.0.0b4. `find_connection_file` only searches in '.' and whatever `jupyter_runtime_dir` returns.

However, since I browse to my kernel.json files I have the full path. So I employed the following fix in `_create_client_for_kernel`:
```python3
def _create_client_for_kernel(self, cf, hostname, kf, pw):
    # Verifying if the connection file exists
    cf = osp.abspath(cf)
    cf_path, cf_name = osp.split(cf)
    #cf = osp.basename(cf)
    try:
        find_connection_file(cf_name, path=cf_path)
    ...
```

There's a chance this fix horribly breaks connecting to a kernel without specifying the full path though. It seems that it looks in the current working directory, if I don't use the Browse button or specify a full path.

For completeness' sake: my remote kernel files live in `~/.ipython/profile_default/security/kernel-*.json`. Maybe this directory should be added to the paths searched? But maybe this is more of a problem for the Jupyter people. 


Keep up the great work, I like 3.0 so far :+1: :)
